cadc-sia/src/main/java/ca/nrc/cadc/sia2/impl/ServiceAvailability.java

```diff
@@ -187,18 +195,20 @@ public static String getTapURI() {
     }
 
     private static String getTapURIProperty() throws IOException {
-        URL url = SiaRunner.class.getResource(CONFIG_FILE_NAME);
-        if (url == null) {
-            url = SiaRunner.class.getResource("/" + CONFIG_FILE_NAME);
-        }
+        // Try to get from a file on disk first.
+        final PropertiesReader propertiesReader = new PropertiesReader(CONFIG_FILE_NAME);
 
-        if (url == null) {
-            final PropertiesReader propertiesReader = new PropertiesReader(CONFIG_FILE_NAME);
+        if (propertiesReader.canRead()) {
             final MultiValuedProperties multiValuedProperties = propertiesReader.getAllProperties();
             final List<String> props = (multiValuedProperties == null)
-                    ? new ArrayList<String>() : multiValuedProperties.getProperty(CONFIG_TAP_URI_KEY);
+                                       ? new ArrayList<String>() : multiValuedProperties.getProperty(
+                    CONFIG_TAP_URI_KEY);
             return props.isEmpty() ? null : props.get(0);
         } else {
+            URL url = SiaRunner.class.getResource(CONFIG_FILE_NAME);
+            if (url == null) {
+                url = SiaRunner.class.getResource("/" + CONFIG_FILE_NAME);
+            }
```

Comment:
Not so fond of the 'try both of these' approach... isn't there only one correct way to do it?  (with or without a slash?)

Reply:
This exists for Java 8 and Java 11 support.  Java 9 (and later) requires the leading slash.

Reply:
There is a method in FileUtil (cadc-util) that does something like this... it should be used (and fixed if necessary) so that when we go to java 9+ and/or someone decides to change classloader semantics we only have one  bug to fix. Then the swugly is only in one place... a comment to explain that ugliness will help prevent it from being broken.

Reply:
OK.  Will change.